library/common/engine.cc

```diff
@@ -94,14 +95,10 @@ Engine::~Engine() {
 }
 
 void Engine::recordCounter(std::string elements, uint64_t count) {
-  if (server_) {
+  if (server_ && client_scope_) {
     server_->dispatcher().post([this, elements, count]() -> void {
-      static const std::string client = "client";
-      absl::string_view prefix{client};
       absl::string_view dynamic_elements{elements};
-      Stats::Utility::counterFromElements(server_->serverFactoryContext().scope(),
-                                          {prefix, dynamic_elements})
-          .add(count);
+      Stats::Utility::counterFromElements(*client_scope_, {dynamic_elements}).add(count);
```

Comment:
I would not have expected this to work, but I see the problem. I meant to make DynamicName an `explicit` but allow the implicit conversion from string literal to string_view.

Can you change this to:
```
  Stats::Utility::counterFromElements(*client_scope_, {Stats::DynamicName(elements)}).add(count);
```
I'm going to write a quick PR to add the `explicit` qualifier to the DynamicName constructor which would make the form that you have here invalid.

The reason is that it is an anti-pattern to have a ton of DynamicName objects (due to string bloat) so I want to be able to search for them.

Of course the better solution is really to get callers to create a Counter and keep a handle to it, rather than doing the name lookup all the time, but if you can't do that then please add the explicit qualifier.

Then also you will not need the string_view declaration above.